File: src/main/java/com/example/estantevirtual/model/Book.java
package com.example.estantevirtual.model;

import lombok.Getter;
import lombok.Setter;
import org.hibernate.validator.constraints.Length;

import javax.persistence.Entity;
import javax.persistence.Id;
import javax.validation.constraints.Min;
import javax.validation.constraints.NotNull;
import java.io.Serializable;
import java.util.UUID;

@Entity
@Getter
@Setter
public class Book implements Serializable {

    @Id
    private UUID id;

    @NotNull(message = "Title is required")
    @Length(min = 3, max = 100, message = "Book title must be between {min} and {max} characters")
    private String title;

    @NotNull(message = "Author is required")
    @Length(min = 3, max = 35, message = "Author name must be between {min} and {max} characters")
    private String author;

    @NotNull(message = "Number of pages is required")
    @Min(value = 1, message = "Number of pages must be greater than 0")
    private Integer pages;
}

File: src/main/kotlin/com/estantevirtual/model/Book.kt
package com.estantevirtual.model

import lombok.Getter
import lombok.Setter
import org.hibernate.validator.constraints.Length
import java.io.Serializable
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.Id
import javax.validation.constraints.Min
import javax.validation.constraints.NotNull

@Entity
@Getter
@Setter
class Book(
    isbn: String,
    title: String,
    author: String,
    pages: Int
) : Serializable {

    @Id
    @Column(name = "isbn")
    var isbn: String? = isbn

    @NotNull(message = "Title is required")
    @Length(min = 3, max = 100, message = "Book title must be between {min} and {max} characters")
    var title: String? = title

    @NotNull(message = "Author is required")
    @Length(min = 3, max = 35, message = "Author name must be between {min} and {max} characters")
    var author: String? = author

    @NotNull(message = "Number of pages are required")
    @Min(value = 1, message = "Number of pages must be greater than 0")
    var pages: Int? = pages
}
